### ISSUE TITLE
Im getting the following error: Parse error: syntax error, unexpected '?', expecting variable (T_VARIABLE)

### ISSUE THREAD

**Issue by Jatapiaro** [doctrine/dbal#3240]
**Symfony version(s) affected**: Latest

**Description**  
On production with PHP Version => 7.2.2-3+ubuntu16.04.1+deb.sury.org+1 I got the following error:
Parse error: syntax error, unexpected '?', expecting variable (T_VARIABLE)

**How to reproduce**  
Just try to access any route on laravel. You need to include `doctrine/dbal`.

**Possible Solution**  
The error throws the line 79 of the code.

```
 /**
     * @throws InvalidArgumentException If a locale contains invalid characters
     */
    public function __construct(?string $locale, MessageFormatterInterface $formatter = null, string $cacheDir = null, bool $debug = false)
    {
        $this->setLocale($locale);
 
        if (null === $formatter) {
            $formatter = new MessageFormatter();
        }
 
        $this->formatter = $formatter;
        $this->cacheDir = $cacheDir;
        $this->debug = $debug;
    }
```

That line corresponds to the method declaration, why does it have a `?string`, I just removed that `?` and everything works fine.

Also I have this stackoverflow issue if you need more information: https://stackoverflow.com/questions/51698746/how-to-fix-a-syntax-error-on-symfony-component-raised-after-composer-require-doc?noredirect=1#comment90359711_51698746

**Comment by Ocramius** [comment]
?string means that the parameter is nullable. You probably have different
PHP versions installed for web and CLI.

**Comment by Jatapiaro** [comment]
Yes, On local I have 7.1. But how does that affects the production server? I added the composer.lock also.

**Comment by Ocramius** [comment]
Production is probably running 7.0 for web and 7.2.2-sury for CLI.

On 6 Aug 2018 02:49, "Jatapiaro" <notifications@github.com> wrote:

Yes, On local I have 7.1. But how does that affects the production server?
I added the composer.lock also.

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/doctrine/dbal/issues/3240#issuecomment-410561276>, or mute
the thread

**Comment by Jatapiaro** [comment]
I run `php -i` on the server and got this: ```Core

PHP Version => 7.2.2-3+ubuntu16.04.1+deb.sury.org+1```

**Comment by Ocramius** [comment]
Yes, that would be the CLI version: what does it say if you upload a file
with `<?php phpinfo();`?

**Comment by Jatapiaro** [comment]
You are right, the web version says `PHP Version 7.0.28-0ubuntu0.16.04.1`

**Comment by wonbin2011** [comment]
yes  the php version results  this
